examples/cargo_without_cmake/build.rs

```diff
@@ -14,7 +14,14 @@ fn main() {
         // - Qt Qml is linked by enabling the qt_qml Cargo feature (default).
         // - Qt Qml requires linking Qt Network on macOS
         .qt_module("Network")
-        .qml_module("com.kdab.cxx_qt.demo", 1, 0, &["src/cxxqt_object.rs"])
+        .qml_module(
+            "com.kdab.cxx_qt.demo",
+            1,
+            0,
+            &["src/cxxqt_object.rs"],
+            &["qml/main.qml"],
+            &[] as &[&str; 0],
```

Comment:
I'm not so happy with this API requiring this type disambiguation boilerplate to pass nothing... suggestions?

Reply:
Maybe use `Option<&[impl AsRef<Path>]>` for this parameter?

Reply:
Ugh that's unfortunate :thinking: As even with option then you need to write `None` or `Some(&[...])` right ?

I still thing long term we should maybe have a builder pattern as an alternate API too as it lends itself better to these scenarios where you need optionals.

Something like
```rust
.qml_module(QmlModuleBuilder::new("com.kdab.cxx_qt.demo", 1, 0)
  .rs(&["src/cxxqt_object.rs"])
  .qml(&["qml/main.qml"])
  .build()
)
```

Then if you know you have all of them you could still do something like this with the resultant struct
```rust
.qml_module(QmlModule {
   uri: "com.kdab.cxx_qt.demo",
   version_major: 1,
   version_minor: 0,
   rs_files: &["src/cxxqt_object.rs"],
   qml_files: &["qml/main.qml"],
   qrc_files: &[],
})
```

But maybe this is also due to me not liking functions with lots of arguments that are of the same type :')

Reply:
To expand on the second idea, you could even have a non-exhaustive struct, as long as there's a default impl for it:
```rust
.qml_module(QmlModule {
    uri: "...",
    qml_files: &[...],
    ..Default::default()
})
```
vulkano uses this pattern extensively and I find it quite readable, as it basically acts as named parameters for a function.

If you can't have a `default`, you can even use this pattern in combination with some kind of other `new` function, though that's starting to become less readable.
```rust
.qml_module(QmlModule {
    rs_files: &[...],
    ..QmlModule::named("com.kdab.cxx_qt.demo", 1, 0)
})
```

Reply:
The second one there looks very interesting!

Reply:
> To expand on the second idea, you could even have a non-exhaustive struct, as long as there's a default impl for it:

I implemented this. I like how `Default::default()` allows skipping specifying optional fields.

This approach has an ugly downside though. To be flexible and accept whatever impls `AsRef<Path>` in the QmlModule struct, CxxQtBuilder needs some generics. When build.rs has a call to CxxQtBuilder::qml_module, this works fine. But, when there is no QML module specified in build.rs, it gets ugly with needing to specify generics for CxxQtBuilder with a turbofish:
```rust
CxxQtBuilder::<&str, &str>::new().file("src/lib.rs").build();
```

Reply:
> This approach has an ugly downside though.

I resolved this by using a separate QmlModule struct in the API than the private Vec owned by CxxQtBuilder. Now we have this nice public API:
```rust
         .qml_module(QmlModule {
            uri: "com.kdab.cxx_qt.demo",
            rust_files: &[
                "src/containers.rs",
                "src/custom_base_class.rs",
                "src/custom_parent_class.rs",
                "src/invokables.rs",
                "src/multiple_qobjects.rs",
                "src/nested_qobjects.rs",
                "src/serialisation.rs",
                "src/signals.rs",
                "src/singleton.rs",
                "src/properties.rs",
                "src/threading.rs",
                "src/types.rs",
                "src/uncreatable.rs",
            ],
            qml_files: &[
                "../qml/main.qml",
                "../qml/pages/ContainersPage.qml",
                "../qml/pages/CustomBaseClassPage.qml",
                "../qml/pages/CustomParentClassPage.qml",
                "../qml/pages/InvokablesPage.qml",
                "../qml/pages/MultipleQObjectsPage.qml",
                "../qml/pages/NestedQObjectsPage.qml",
                "../qml/pages/PropertiesPage.qml",
                "../qml/pages/SerialisationPage.qml",
                "../qml/pages/SignalsPage.qml",
                "../qml/pages/SingletonPage.qml",
                "../qml/pages/ThreadingPage.qml",
                "../qml/pages/TypesPage.qml",
            ],
            ..Default::default()
        })
```

and users do not need to explicitly specify generics.

Reply:
Nice :-D